discourse-doctor

```diff
@@ -0,0 +1,374 @@
+  read_config "DISCOURSE_DEVELOPER_EMAILS"
+  DEVELOPER_EMAILS=$read_config_result
+  echo   DEVELOPER_EMAILS=$DEVELOPER_EMAILS| tee -a  $LOG_FILE
+  read_config "DISCOURSE_SMTP_PASSWORD"
+  SMTP_PASSWORD=$read_config_result
+  echo   SMTP_PASSWORD=$read_config_result| tee -a  $LOG_FILE
+  read_config "DISCOURSE_SMTP_PORT"
+  SMTP_PORT=$read_config_result
+  echo   SMTP_PORT=$read_config_result| tee -a  $LOG_FILE
+  read_config "DISCOURSE_SMTP_USER_NAME"
+  SMTP_USER_NAME=$read_config_result
+  echo   SMTP_USER_NAME=$read_config_result| tee -a  $LOG_FILE
+  read_config "LETSENCRYPT_ACCOUNT_EMAIL"
+  letsencrypt_account_email=$read_config_result
+  echo "LETSENCRYPT_ACCOUNT_EMAIL=$letsencrypt_account_email"
+  if [ -z "$letsencrypt_account_email" ]
+  then
+    echo "LETSENCRYPT_ACCOUNT_EMAIL is empty" | tee -a $LOG_FILE
+  else
+    echo "LETSENCRYPT_ACCOUNT_EMAIL is set" | tee -a $LOG_FILE
+  fi
+}
+
+get_plugins() {
+  echo -e "\n\n==================== PLUGINS ====================" | tee -a $LOG_FILE
+  grep git containers/$app_name.yml | tee -a $LOG_FILE
+
+  grep git containers/$app_name.yml > /tmp/$PPID.grep
+
+  if grep -cv "github.com/discourse" /tmp/$PPID.grep > /dev/null
+  then
+    echo -e "\nWARNING:" | tee -a $LOG_FILE
+    echo You have what appear to be non-official plugins. | tee -a $LOG_FILE
+    echo "If you are having trouble, you should disable them and try rebuilding again." | tee -a $LOG_FILE
+  fi
+
+}
+
+dump_yaml() {
+  echo -e "\n\n==================== YML DUMP ====================" | tee -a $LOG_FILE
+  echo Dumping $app_name.yml | tee -a $LOG_FILE
+  echo -e "\n\n" | tee -a $LOG_FILE
+}
+
+##
+## read a variable from the config file and stick it in read_config_result
+##
+read_config() {
+  config_line=`egrep "^  #?$1:" $web_file`
+  read_config_result=`echo $config_line | awk  --field-separator=":" '{print $2}'`
+  read_config_result=`echo $read_config_result | sed "s/^\([\"']\)\(.*\)\1\$/\2/g"`
+}
+
+##
+## call rake emails:test inside the container
+##
+email_test() {
+  echo -e "\n\n==================== MAIL TEST ===================="
+  echo "For a robust test, get an address from http://www.mail-tester.com/ , but you "
+  echo "can just send a test message to yourself."
+  EMAIL=`echo $DEVELOPER_EMAILS |cut -d , -f 1`
+  read -p "Email address for mail test? ('n' to skip) [$EMAIL]: " new_value
+  if [ ! -z "$new_value" ]
+  then
+    EMAIL="$new_value"
+  fi
+  if [ "$new_value" != "n" ] && [ "$new_value" != "N" ]
+  then
+    echo "Sending mail to $EMAIL. . . "| tee -a $LOG_FILE
+    sudo docker exec -w /var/www/discourse -i $app_name rake emails:test[$EMAIL] |& tee -a $LOG_FILE
+  else
+    echo "Mail test skipped."
+  fi
+}
+
+##
+## Get container name
+##
+
+docker_check() {
+  if [ -f /var/discourse/containers/app.yml ]
+  then
+    echo "YML=app.yml"
+    app_name=app
+    data_name=app
+    web_file=containers/$app_name.yml
+    data_file=containers/$app_name.yml
```

Comment:
where is this variable `data_file` being used ?

Reply:
It's not, but if we add a Postgres version check in the future, we have a hook to be able to know how to get into the data container in the event that they have a separate data container that's called data.

If you want, I'll remove it.